ManagedHooksManager/HooksManager.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.WindowsRuntime;
+
+namespace ManagedHooksManager
+{
+    public sealed class HooksManager
+    {
+        private readonly ConcurrentDictionary<string, string> _winRtTypesRegistry;
+        private readonly string _executingAssemblyDir;
+
+        private ApiHook<NativeApi.RoGetActivationFactory> _roGetActivationFactoryHook;
+        private ApiHook<NativeApi.RoResolveNamespace> _roResolveNamespaceHook;
+
+        public HooksManager()
+        {
+            _winRtTypesRegistry = new ConcurrentDictionary<string, string>();
+            _executingAssemblyDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+        }
+
+        public void RegisterWinRtType(string assemblyQualifiedWinRtTypeName)
+        {
+            var winRtType = Type.GetType(assemblyQualifiedWinRtTypeName, throwOnError: true);
+            RegisterWinRtType(winRtType);
+        }
+
+        public void RegisterWinRtType<T>()
+        {
+            RegisterWinRtType(typeof(T));
+        }
+
+        public void RegisterWinRtType(Type winRtType)
+        {
+            if (winRtType == null)
+                throw new ArgumentNullException(nameof(winRtType));
+
+            var implementationPath = Path.ChangeExtension(winRtType.Assembly.Location, "dll");
+
+            _winRtTypesRegistry.AddOrUpdate(
```

Comment:
This registration idea is mostly a shortcut to get implementation files for WinRT types. It assumes that your .dll implementation file has the same name as your .winmd metadata file (which is usually the default case). It also ensures you aknowledge all the native components you are using in your code. Of course, you don't need to register system components (that is, from Windows.* namespace) as they are successfully resolved by the system itself.